src/rc_allocator.c

```diff
@@ -262,9 +301,17 @@ rc_allocator_init(rc_allocator        *al,
    al->heap_handle = hh;
    al->heap_id     = hid;
 
+   rc = rc_allocator_valid_config(cfg);
+   if (!SUCCESS(rc)) {
+      return rc;
+   }
    platform_assert(cfg->io_cfg->page_size % 4096 == 0);
    platform_assert(cfg->capacity
-                   == cfg->io_cfg->extent_size * cfg->extent_capacity);
+                      == cfg->io_cfg->extent_size * cfg->extent_capacity,
```

Comment:
This assertion is now redundant.  Delete.

Reply:
Deleted. Also L308 came out. As did this check below, which is now caught by a real error message:

```
-   platform_assert(cfg->capacity
-                   == cfg->io_cfg->page_size * cfg->page_capacity);
```